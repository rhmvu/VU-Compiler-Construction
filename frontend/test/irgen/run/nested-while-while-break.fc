extern int printf(char[] format, ...);

int main()
{
    int i = 0;
    while (i < 9) {
        i += 1;
        //1 and 2 adn 3 only
        do {
            do {
                do {
                    printf("deepest dowhile int:%d\n", i);
                    break;
                    printf("WRONG: deepest dowhile int:%d\n", i);
                } while (1==1);
                break;
            } while (1==1);
            i += 1;
            if (i < 4) {
                continue;
            }
            else {
                break;
                printf("FAILFAILFAIL");
            }
        } while (1==1);
	printf("%d\n", i);
        if (i > 7)
            continue;
    }
	return 0;
}
