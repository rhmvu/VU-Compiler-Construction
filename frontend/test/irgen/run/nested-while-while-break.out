deepest dowhile int:1
deepest dowhile int:2
deepest dowhile int:3
4
deepest dowhile int:5
6
deepest dowhile int:7
8
deepest dowhile int:9
10

